Common/Data/Consolidators/VolumeRenkoConsolidator.cs

```diff
@@ -0,0 +1,289 @@
+        /// </summary>
+        /// <remarks>Protected for testing</remarks>
+        protected DateTime OpenOn;
+
+        /// <summary>
+        /// Value of consolidate open.
+        /// </summary>
+        /// <remarks>Protected for testing</remarks>
+        protected decimal OpenRate;
+
+        /// <summary>
+        /// Size of the consolidated bar.
+        /// </summary>
+        /// <remarks>Protected for testing</remarks>
+        protected decimal BarSize;
+
+        /// <summary>
+        /// Gets the kind of the bar
+        /// </summary>
+        public RenkoType Type => RenkoType.Wicked;
+
+        /// <summary>
+        /// Gets a clone of the data being currently consolidated
+        /// </summary>
+        public IBaseData WorkingData => _currentBar?.Clone();
+
+        /// <summary>
+        /// Gets the type consumed by this consolidator
+        /// </summary>
+        public Type InputType => typeof(IBaseData);
+
+        /// <summary>
+        /// Gets <see cref="VolumeRenkoBar"/> which is the type emitted in the <see cref="IDataConsolidator.DataConsolidated"/> event.
+        /// </summary>
+        public Type OutputType => typeof(VolumeRenkoBar);
+
+        /// <summary>
+        /// Gets the most recently consolidated piece of data. This will be null if this consolidator
+        /// has not produced any data yet.
+        /// </summary>
+        public IBaseData Consolidated
+        {
+            get { return _consolidated; }
+            private set { _consolidated = value; }
+        }
+
+        /// <summary>
+        /// Event handler that fires when a new piece of data is produced
+        /// </summary>
+        public event EventHandler<VolumeRenkoBar> DataConsolidated;
+
+        /// <summary>
+        /// Event handler that fires when a new piece of data is produced
+        /// </summary>
+        event DataConsolidatedHandler IDataConsolidator.DataConsolidated
+        {
+            add { _dataConsolidatedHandler += value; }
+            remove { _dataConsolidatedHandler -= value; }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="VolumeRenkoConsolidator"/> class using the specified <paramref name="barSize"/>.
+        /// </summary>
+        /// <param name="barSize">The constant volume size of each bar</param>
+        public VolumeRenkoConsolidator(decimal barSize)
+        {
+            BarSize = barSize;
+        }
+
+        /// <summary>
+        /// Updates this consolidator with the specified data
+        /// </summary>
+        /// <param name="data">The new data for the consolidator</param>
+        public void Update(IBaseData data)
+        {
+            var rate = data.Price;
+            var volume = _volumeLeftOver;
+
+            if (data.GetType() == typeof(QuoteBar))
+            {
+                throw new ArgumentException("VolumeRenkoConsolidator() must be used with TradeBar or Tick data.");
+            }
```

Comment:
Ideally this is the `else` case? we can avoid this comparison in normal use cases. We should call `data.GetType()` once at the top